core/src/main/scala/org/apache/spark/deploy/StandaloneSparkHadoopUtil.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy
+
+import java.security.PrivilegedAction
+
+import org.apache.hadoop.mapred.JobConf
+import org.apache.hadoop.security.UserGroupInformation
+import org.apache.spark.{SparkEnv, Logging}
+import sun.security.jgss.krb5.Krb5InitCredential
+
+import scala.sys.process.Process
+
+private[spark] class StandaloneSparkHadoopUtil extends SparkHadoopUtil {
+
+  val hadoopSecurityAuthenticationKey = "hadoop.security.authentication"
+  val principalKey = "dfs.namenode.kerberos.principal"
+  val keytabKey = "dfs.namenode.keytab.file"
+  val securityEnabledButKeyNotDefined = "Hadoop security was enabled, but %s" +
+    "was not set in the hadoop configuration."
+
+  lazy val configuredPrivilegedUGI: UserGroupInformation = {
+    if ("kerberos".equals(conf.get(hadoopSecurityAuthenticationKey, "simple"))) {
+      logInfo("Setting up kerberos to authenticate")
+      getConfValueOrNone(principalKey) match {
+        case Some(principal) =>
+          getConfValueOrNone(keytabKey) match {
+            case Some(keytab) =>
+              // Log in with the HDFS principal. However, this does NOT get the TGT
+              // kinit must be executed manually in spark-env.sh or somewhere similar
+              UserGroupInformation.loginUserFromKeytabAndReturnUGI(principal, keytab)
```

Comment:
@tgravescs if you have any idea it would be helpful (personally not really keyed in on how this works). @mccheah my understanding was that in general if you have a keytab you shouldn't need to run kinit ever in order for Hadoop to be able to log in.

You might take a look at how this was done in previous patches that also tried to do something similar:

https://github.com/apache/spark/pull/2320/files#diff-7f5edfa5bc1efcd65bab2f7e0e3320a5R222